internal/proxy/search_util.go

```diff
@@ -168,30 +175,35 @@ func parseSearchInfo(searchParamsPair []*commonpb.KeyValuePair, schema *schemapb
 	} else {
 		groupByInfo := parseGroupByInfo(searchParamsPair, schema)
 		if groupByInfo.err != nil {
-			return nil, 0, groupByInfo.err
+			return &SearchInfo{planInfo: nil, offset: 0, isIterator: false, parseError: groupByInfo.err}
 		}
 		groupByFieldId, groupSize, groupStrictSize = groupByInfo.GetGroupByFieldId(), groupByInfo.GetGroupSize(), groupByInfo.GetGroupStrictSize()
 	}
 
 	// 6. parse iterator tag, prevent trying to groupBy when doing iteration or doing range-search
 	if isIterator == "True" && groupByFieldId > 0 {
-		return nil, 0, merr.WrapErrParameterInvalid("", "",
-			"Not allowed to do groupBy when doing iteration")
+		return &SearchInfo{planInfo: nil, offset: 0, isIterator: false, parseError: merr.WrapErrParameterInvalid("", "",
+			"Not allowed to do groupBy when doing iteration")}
 	}
 	if strings.Contains(searchParamStr, radiusKey) && groupByFieldId > 0 {
-		return nil, 0, merr.WrapErrParameterInvalid("", "",
-			"Not allowed to do range-search when doing search-group-by")
-	}
-
-	return &planpb.QueryInfo{
-		Topk:            queryTopK,
-		MetricType:      metricType,
-		SearchParams:    searchParamStr,
-		RoundDecimal:    roundDecimal,
-		GroupByFieldId:  groupByFieldId,
-		GroupSize:       groupSize,
-		GroupStrictSize: groupStrictSize,
-	}, offset, nil
+		return &SearchInfo{planInfo: nil, offset: 0, isIterator: false, parseError: merr.WrapErrParameterInvalid("", "",
+			"Not allowed to do range-search when doing search-group-by")}
+	}
+
+	return &SearchInfo{
+		planInfo: &planpb.QueryInfo{
+			Topk:            queryTopK,
+			MetricType:      metricType,
+			SearchParams:    searchParamStr,
+			RoundDecimal:    roundDecimal,
+			GroupByFieldId:  groupByFieldId,
+			GroupSize:       groupSize,
+			GroupStrictSize: groupStrictSize,
+		},
+		offset:     offset,
+		isIterator: isIterator == "True",
```

Comment:
Nit: "True" const is friendly to language that boolean const does not start with capital T